LB_D2Q9/dimensionless/cython_dim.pyx

```diff
@@ -0,0 +1,596 @@
+                y = y_list[i]
+
+                old_f0 = f[0, x, y]
+                old_f1 = f[1, x, y]
+                old_f2 = f[2, x, y]
+                old_f3 = f[3, x, y]
+                old_f4 = f[4, x, y]
+                old_f5 = f[5, x, y]
+                old_f6 = f[6, x, y]
+                old_f7 = f[7, x, y]
+                old_f8 = f[8, x, y]
+
+                # Bounce back everywhere!
+                # left right
+                f[1, x, y] = old_f3
+                f[3, x, y] = old_f1
+                # up down
+                f[2, x, y] = old_f4
+                f[4, x, y] = old_f2
+                # up-right
+                f[5, x, y] = old_f7
+                f[7, x, y] = old_f5
+
+                # up-left
+                f[6, x, y] = old_f8
+                f[8, x, y] = old_f6
+
+### Matt Stuff ###
+
+# class Velocity_Inlet_Cylinder(Pipe_Flow_Cylinder):
+#
+#     def __init__(self, *args, inlet_velocity=None, **kwargs):
+#
+#         assert inlet_velocity is not None
+#         self.phys_inlet_velocity = inlet_velocity
+#         self.dim_inlet_velocity = None
+#         self.lb_inlet_velocity = None
+#
+#         super(Velocity_Inlet_Cylinder, self).__init__(**kwargs)
+#
+#
+#     def set_characteristic_length_time(self):
+#         """Necessary for subclassing"""
+#         self.L = self.phys_cylinder_radius
+#         self.T = self.phys_cylinder_radius/self.phys_inlet_velocity
+#
+#         self.dim_inlet_velocity = (self.T/self.L) * self.phys_inlet_velocity
+#
+#     def initialize_grid_dims(self):
+#         """Necessary for subclassing"""
+#
+#         self.lx = int(np.ceil((self.phys_pipe_length / self.L)*self.N))
+#         self.ly = int(np.ceil((self.phys_diameter / self.L)*self.N))
+#
+#         self.nx = self.lx + 1 # Total size of grid in x including boundary
+#         self.ny = self.ly + 1 # Total size of grid in y including boundary
+#
+#         ## Initialize the obstacle mask
+#         self.obstacle_mask = np.zeros((self.nx, self.ny), dtype=np.bool, order='F')
+#
+#         # Initialize the obstacle in the correct place
+#         x_cylinder = self.N * self.phys_cylinder_center[0]/self.L
+#         y_cylinder = self.N * self.phys_cylinder_center[1]/self.L
+#
+#         circle = ski.draw.circle(x_cylinder, y_cylinder, self.N)
+#         self.obstacle_mask[circle[0], circle[1]] = True
+#
+#     def move_bcs(self):
+#         """This is slow; cythonizing makes it fast."""
+#
+#         cdef int lx = self.lx
+#         cdef int ly = self.ly
+#
+#         cdef float u_w = self.lb_inlet_velocity
+#         cdef float u_e = self.lb_inlet_velocity
+#
+#         cdef float[:,:,:] farr = self.f
+#         cdef float[:] rho_w = self.rho[0, 1:ly]
+#         cdef float[:] rho_e = self.rho[lx, 1:ly]
+#
+#         cdef float[:,:,:] f = self.f
+#         cdef int ii
+#
+#         for ii in range(1,ly):
+#             # INLET: imposed velocity of u_w in the x direction and 0 in the y direction
+#
```

Comment:
Something crazy was happening here with rhos, where you were setting rho at the inlet and rho at the outlet in the move_bcs method. This is *extremely* dangerous. You should do this in another class. I deleted the line and tried to fix it, but saw crazy things.